src/storage/src/source/types.rs

```diff
@@ -181,6 +182,39 @@ impl<Key, Value, Time: Timestamp, Diff> SourceMessageType<Key, Value, Time, Diff
     }
 }
 
+#[derive(Clone)]
+pub struct ByteStream {
+    /// The underlying asynchronous stream
+    pub stream: Rc<LocalBoxStream<'static, Vec<u8>>>,
+    /// The expected size of this stream in bytes. This is only an estimation and no assumption
+    /// should be based on the accuracy of this value.
+    pub size_hint: Option<usize>,
```

Comment:
can you just use https://docs.rs/futures/latest/futures/stream/trait.Stream.html#method.size_hint

Reply:
That method is slightly different because it is a size_hint about the number of elements but we're interested in the number of bytes. It seemed a bit of an abuse to use that for MaybeLength

Reply:
Ah I see! `MaybeLength` is probably not long for this world!